File: test/four/app/src/main/java/com/z3d/display/particle/ctrl/SelfRotation.java
package com.z3d.display.particle.ctrl;

import java.util.List;

public class SelfRotation extends BaseAnim {

    public void dataByte(List  va, List arr) {
//        this.beginTime =(float) arr.get(0);
//        float indx001=(float)  arr.get(1);
//        if (indx001== -1) {
//            this.lastTime = Scene_data.MAX_NUMBER;
//        } else {
//            this.lastTime =indx001;
//        }
        super.dataByte(va,arr);
        this.speed =(float)arr.get(2) * 0.1f;
        this.aSpeed =(float)arr.get(3) * 0.1f;
    }
}

File: metalios/iosgl/iosgl/pan3d/display/particle/ctrl/SelfRotation.m
//
//  SelfRotation.m
//  iosgl
//
//  Created by zhao on 31/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "SelfRotation.h"
#import "Scene_data.h"

@implementation SelfRotation


-(void)dataByte:(NSArray<NSNumber*>*)va arr:(NSArray<NSNumber*>*)arr;
{
    self.beginTime = [arr[0] floatValue];
    if ([arr[1]floatValue] == -1) {
        self.lastTime = [Scene_data default] .MAX_NUMBER;
    } else {
        self.lastTime = [arr[1]floatValue];
    }
    self.speed = [arr[2]floatValue] * 0.1;
    self.aSpeed = [arr[3]floatValue] * 0.1;
}
/*
public dataByte(va: Array<any>, arr: Array<any>): void {
    this.beginTime = arr[0]
    if (arr[1] == -1) {
        this.lastTime = Scene_data.MAX_NUMBER;
    } else {
        this.lastTime = arr[1];
    }
    this.speed = arr[2] * 0.1;
    this.aSpeed = arr[3] * 0.1;

}
*/
@end
